### ISSUE TITLE
Rename white pattern / black pattern to e.g. pattern / bypass pattern

### ISSUE THREAD

**Issue by murb** [foxyproxy/firefox-extension#81]
Please consider renaming black patterns and white patterns to something more inclusive and clearer. Besides that associating black with not passing and white with passing, is kind of offensive, I'd also think that the function of the two lists can be more clearly communicated when being renamed to e.g.

Proxy patterns and By pass patterns

Other projects have undergone similar changes. While a typical replacement seems to be blocklist / allowlist 'bypass' might be more suitable in the context of a proxy service(?) I'm not a native English speaker, maybe there is even a better term?

* https://github.com/rails/rails/issues/33677
* https://lkml.org/lkml/2020/6/9/1468
* https://www.theregister.com/2019/09/03/chromium_microsoft_offensive/

**Comment by ericjung** [comment]
Happy Juneteenth. Please read response [here](https://blog.getfoxyproxy.org/2020/06/19/a-black-white-coding-issue/).

TL;DR pull requests encouraged if you want this done faster. We will likely use **green and red patterns** to replace **white and black** in keeping with the colors analogy.

**Comment by murb** [comment]
Thanks for your response! While red and green seem fine to me, dropping color-coding altogether as it still adds another level of abstraction; you still have to explain it... "green patterns" are patterns that will be fed into the proxy, whereas "red patterns" will bypass the proxy... why not state "proxy patterns" and "bypass patterns" (or "exception patterns")?

**Comment by dmrzzz** [comment]
1. Thanks for taking this seriously!
2. From a user experience perspective, every single one of @murb's functional naming suggestions above is much clearer than green/red.  Two more possible options (just in case they happen to resonate more): "match patterns" and "exclude patterns".
3. The blog post claims that FoxyProxy does not use "whitelist", but that's not accurate; the phrase "Add whitelist pattern to match all URLs" appears in at least two places in the UI (the screenshot shown in the blog entry, and also when configuring a new proxy)
4. Thanks for FoxyProxy in general, it's a fantastic add-on.

**Comment by xurizaemon** [comment]
Agree with @murb analysis above that reducing abstraction is better. "proxy" and "bypass" (or perhaps "proxy" and "direct"? per [proxy.pac usage](https://developer.mozilla.org/en-US/docs/Web/HTTP/Proxy_servers_and_tunneling/Proxy_Auto-Configuration_(PAC)_file)) seem clearer than "red" / "green", considering "red means stop" but in this context red would mean preventing proxy / go directly.

As above, thanks for FoxyProxy!

**Comment by eean** [comment]
folks are ignoring that you can have more than one proxy so really all patterns are proxy patterns. I see why folks are bouncing off of red/green since stoplights don't filter, so it's an imperfect analogy. But it seems like a fine analogy to use.

though an actual filter analogy might be better, like catch patterns and release patterns (analogy to fishing) ?

**Comment by erosman** [comment]
What about a simple Include/Exclude?!

**Comment by ericjung** [comment]
yes, include/exclude seems the simplest.

**Comment by erosman** [comment]
Done for [FoxyProxy v8.0](https://github.com/foxyproxy/browser-extension) (once released).
